tests/test_is_instance.py

```diff
@@ -44,6 +44,10 @@ def test_typed_tuples():
     assert not isinstance(('cake', 'pie', 42), (str, str, int))
     assert not is_instance(('cake', 'pie', 42), (str, str, int))
 
+    assert is_instance((), tuple[int, ...])
```

Comment:
I'm gonna wrap these in a TODO after committing this so they don't execute yet, but I like where you're going with this so I'll just commit it as is for now. :)